src/device.rs

```diff
@@ -409,8 +436,8 @@ pub trait DeviceV1_0 {
                                        pipelines.as_mut_ptr());
         pipelines.set_len(create_infos.len());
         match err_code {
-            vk::Result::Success => Ok(pipelines),
-            _ => Err(err_code),
+            vk::Result::Success => (pipelines, Ok(())),
+            _ => (pipelines, Err(err_code)),
```

Comment:
I am a bit confused after reading https://www.khronos.org/registry/vulkan/specs/1.0/man/html/vkCreateGraphicsPipelines.html

It seems that the function can only fail if the host or the device is oom. How can you recover pipelines that failed to build? I assumed that no pipeline would be returned if an error occurred.